src/cursor/cur_stat.c

```diff
@@ -357,6 +357,7 @@ __curstat_conn_init(WT_SESSION_IMPL *session, WT_CURSOR_STAT *cst)
 	 * Optionally clear the connection statistics.
 	 */
 	__wt_conn_stat_init(session);
+	memset(&cst->u.conn_stats, 0, sizeof(WT_CONNECTION_STATS));
```

Comment:
Would this  clear the value for the `no_clear` statistics too?

Reply:
Yes, it is. But only clear the aggregated conn stats in the cursor.
Global conn stat is cleared at `__wt_stat_coninection_clear_all()` considering the `no_clear` flag, which is at right after this `memset`,

Reply:
Hello @keithbostic, Can you please let me know if there is some reason for using "+=" operator rather than "=" operator inside the `__wt_stat_connection_aggregate()` function? Or can it be simply changed to "=" operator?

Reply:
> Can you please let me know if there is some reason for using "+=" operator rather than "=" operator inside the `__wt_stat_connection_aggregate()` function? Or can it be simply changed to "=" operator?

I'm sorry, but I just don't remember!

My guess is it could be "=", but "+=" makes the auto-generation of the code  simpler because it's all the same, and there was some chance there would be a need for aggregation from multiple connections someday. The only downside I can see is it's marginally slower, but I can't imagine that matters here.

Reply:
Thank you for your answer. Now I can understand it!